### PR TITLE
Javascript2 week2/ciprian

### PR DESCRIPTION
**Pull request by CiprianP84**
I created searching for keyword functionality, sorting by multiple properties and then some specific sorting function - like sorting movies over 2 hours, or by their Rotten Tomato score. 
From what I understood, the actual implementation of those functions is a task for one of the later weeks so I am testing the functionality via console.logs. 
Let me know if I misunderstood something.
Also included some design improvements for mobile view.

### CODE REVIEW

File: searchAndSort.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+// Function to filter by age rating
+function filterMoviesByAgeRating(ageRating) {
+  return moviesDatabase.filter(movie => 
+    movie.age_rating === ageRating
+  );
+}
+
+// Function to show movies over 120min
+const moviesOver2Hours = moviesDatabase
+.filter(movie => movie.movie_duration > 120)
```
Comment (vdaguenet) [inline searchAndSort.js]:
```suggestion
  .filter(movie => movie.movie_duration > 120)
```
Add indentation when you chain functions on a different line

**Review by vdaguenet — APPROVED** [review]
Excellent work!

> From what I understood, the actual implementation of those functions is a task for one of the later weeks so I am testing the functionality via console.logs.
Let me know if I misunderstood something.

You are correct. The brief does not ask you to search and filter based on the user input before JS3 week 2. It's a bit odd in my opinion. So feel free to bind your filtering and sorting functions to the user inputs if you have time. But I will not request changes if you don't do it.

**Comment by CiprianP84** [comment]
Hi, 
Thanks a lot for your feedback.
I implemented the search functionality within my app and fixed the indentation. 
I'll wait with the sorting as I am not yet sure how to integrate it.

File: movieApp.js [hunk 2]
```diff
@@ -1,13 +1,56 @@
...
+    }
+  });
+}
+
+// Mobile search
+function setupMobileSearch() {
+  const searchInputMobile = document.getElementById('search-input-mobile');
+  const mobileMenu = document.getElementById('mobile-menu');
+
+  searchInputMobile.addEventListener('input', () => {
```
Comment (vdaguenet) [inline movieApp.js:31]: You could have a shared event handler with the desktop search by using `event.target` like this:
```js
function onSearch(event) {
  const keyword = event.target.value;
  const filteredMovies = searchMovies(keyword);
  showMovies(filteredMovies);
}
searchInputMobile.addEventListener('input', onSearch);
// ...
searchInput.addEventListener('input', onSearch);
```

**Review by vdaguenet — APPROVED** [review]

**Pull request merged**